google/resource_compute_attached_disk_test.go

```diff
@@ -0,0 +1,218 @@
+				),
+			},
+		},
+	})
+}
+
+func TestAccAttachedDisk_defaults(t *testing.T) {
+	t.Parallel()
+
+	diskName := acctest.RandomWithPrefix("tf-test")
+	instanceName := acctest.RandomWithPrefix("tf-test")
+	importID := fmt.Sprintf("%s/us-central1-a/%s:%s", getTestProjectFromEnv(), instanceName, diskName)
+
+	resource.Test(t, resource.TestCase{
+		PreCheck:     func() { testAccPreCheck(t) },
+		Providers:    testAccProviders,
+		CheckDestroy: nil,
+		Steps: []resource.TestStep{
+			resource.TestStep{
+				Config: testAttachedDiskResource(diskName, instanceName) + testAttachedDiskResourceAttachmentDefaults(),
+			},
+			resource.TestStep{
+				ResourceName:      "google_compute_attached_disk.test",
+				ImportStateId:     importID,
+				ImportState:       true,
+				ImportStateVerify: true,
+			},
+		},
+	})
+
+}
+
+func TestAccAttachedDisk_count(t *testing.T) {
+	t.Parallel()
+
+	diskPrefix := acctest.RandomWithPrefix("tf-test")
+	instanceName := acctest.RandomWithPrefix("tf-test")
+	count := 2
+
+	resource.Test(t, resource.TestCase{
+		PreCheck:     func() { testAccPreCheck(t) },
+		Providers:    testAccProviders,
+		CheckDestroy: nil,
+		Steps: []resource.TestStep{
+			resource.TestStep{
+				Config: testAttachedDiskResourceCount(diskPrefix, instanceName, count),
+				Check: resource.ComposeTestCheckFunc(
+					testAccAttachedDiskContainsManyDisks(instanceName, count),
+				),
+			},
+		},
+	})
+
+}
+
+func testAccAttachedDiskIsNowDetached(instanceName, diskName string) resource.TestCheckFunc {
+	return func(s *terraform.State) error {
+		config := testAccProvider.Meta().(*Config)
+
+		instance, err := config.clientCompute.Instances.Get(getTestProjectFromEnv(), "us-central1-a", instanceName).Do()
+		if err != nil {
+			return err
+		}
+
+		ad := findDiskByName(instance.Disks, diskName)
+		if ad != nil {
+			return fmt.Errorf("compute disk is still attached to compute instance")
+		}
+
+		return nil
+	}
+}
+
+func testAccAttachedDiskContainsManyDisks(instanceName string, count int) resource.TestCheckFunc {
+	return func(s *terraform.State) error {
+		config := testAccProvider.Meta().(*Config)
+
+		instance, err := config.clientCompute.Instances.Get(getTestProjectFromEnv(), "us-central1-a", instanceName).Do()
+		if err != nil {
+			return err
+		}
+
+		// There will be 1 extra disk because of the compute instance's boot disk
+		if (count + 1) != len(instance.Disks) {
+			return fmt.Errorf("expected %d disks to be attached, found %d", count+1, len(instance.Disks))
+		}
+
+		return nil
+	}
+}
+
+func testAttachedDiskResourceAttachment() string {
+	return fmt.Sprintf(`
+resource "google_compute_attached_disk" "test" {
+	disk = "${google_compute_disk.test1.self_link}"
+	instance = "${google_compute_instance.test.self_link}"
+}
+	`)
+}
+
+func testAttachedDiskResourceAttachmentDefaults() string {
+	return fmt.Sprintf(`
+resource "google_compute_attached_disk" "test" {
+	disk = "${google_compute_disk.test1.self_link}"
```

Comment:
it's not a huge deal but it usually looks better when all the `=` are aligned within a section (that's what terraform fmt does on .tf files)